### PR TITLE
Create bash script that installs and configures nginx reverse proxy

### PR DESCRIPTION
**Pull request by tyrotoxin**
X-Pack trial license expires in 30 days, so we just want to install NGINX reverse proxy on client nodes that does SSL termination.

- [x] Create bash script that works
- [x] Add parameters and call the script from templates
- [x] Configure external LB to use port 443 instead of 9200
- [x] Test and verify that everything works

### CODE REVIEW

**Comment by tyrotoxin** [comment]
@mhintzke13 , @AlexBauer-Skykick, this is ready for review

File: src/scripts/nginx-install.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+write_site_config()
+{
+  rm $DEFAULT_SITE_PATH
+
+  echo "server {
+	listen $NGINX_PORT ssl default_server;
+	listen [::]:$NGINX_PORT ssl default_server;
+	location / {
+		proxy_pass $ES_LOCAL_ADDRESS;
+		auth_basic "ElasticSearch!";
```
Comment (mhintzke13) [inline src/scripts/nginx-install.sh]: Is this `auth_basic` value always going to be this hardcoded string?

File: src/settings/ubuntuSettings.json [hunk 2]
```diff
@@ -132,7 +140,7 @@
             "fileUris": "[variables('ubuntuScripts')]"
           },
           "protectedSettings": {
-            "commandToExecute": "[concat('bash elasticsearch-install.sh -y', variables('commonShortOpts'), variables('commonInstallParams'))]"
+            "commandToExecute": "[concat('bash elasticsearch-install.sh -y', variables('commonShortOpts'), variables('commonInstallParams'), '; bash nginx-install.sh', variables('nginxInstallParams'))]"
```
Comment (mhintzke13) [inline src/settings/ubuntuSettings.json]: Nice touch so you don't have to call your nginx install script inside elasticsearch-install and deal with all the params :)

Comment (tyrotoxin) [reply src/settings/ubuntuSettings.json]: This works in bash, and hope this works from ARM templates :) Need to test

File: src/scripts/nginx-install.sh (hunk 1 above)
Comment (tyrotoxin) [reply src/scripts/nginx-install.sh]: This is just a string that shows in a creds prompt. It can be anything, no specific value required.

**Pull request merged**